src/SFA.DAS.EmployerFinance.MessageHandlers.UnitTests/CommandHandlers/DraftExpireAccountFundsCommandHandlerTests.cs

```diff
@@ -0,0 +1,91 @@
+
+            await handler.Handle(message,new TestableMessageHandlerContext(new MessageMapper()));
+
+            expiredFundsRepository.Verify(x=>x.CreateDraft(message.AccountId, It.IsAny<IEnumerable<ExpiredFund>>(), It.IsAny<DateTime>()));
+        }
+
+        [Test, MoqAutoData]
+        public async Task Then_The_Draft_Expired_Funds_Are_Created_From_The_Command_And_Only_Include_Values_Upto_The_Supplied_Date_And_Other_DraftExpiredFunds(
+            DraftExpireAccountFundsCommand message,
+            IDictionary<CalendarPeriod, decimal> expiredFund,
+            LevyFundsIn levyOne,
+            LevyFundsIn levyTwo,
+            PaymentFundsOut paymentOne,
+            PaymentFundsOut paymentTwo,
+            int expiredFundsPeriod,
+            Mock<EmployerFinanceConfiguration> configuration,
+            [Frozen] Mock<ILevyFundsInRepository> levyFundsInRepository,
+            [Frozen] Mock<IPaymentFundsOutRepository> paymentFundsOutRepository,
+            [Frozen] Mock<IExpiredFunds> expiredFunds,
+            [Frozen] Mock<IExpiredFundsRepository> expiredFundsRepository,
+            DraftExpireAccountFundsCommandHandler handler)
+        {
+            //Arrange
+            configuration.Setup(x => x.FundsExpiryPeriod).Returns(expiredFundsPeriod);
+            message.DateTo = DateTime.Now.AddMonths(-1);
+            levyOne.CalendarPeriodYear = message.DateTo.Value.Year;
+            levyOne.CalendarPeriodMonth = message.DateTo.Value.Month;
+            levyTwo.CalendarPeriodYear = DateTime.Now.Year;
+            levyTwo.CalendarPeriodMonth = DateTime.Now.Month;
+            var levyFundsIns = new List<LevyFundsIn> {levyOne, levyTwo};
+            levyFundsInRepository.Setup(x => x.GetLevyFundsIn(message.AccountId))
+                .ReturnsAsync(levyFundsIns);
+
+            paymentOne.CalendarPeriodYear = message.DateTo.Value.Year;
+            paymentOne.CalendarPeriodMonth = message.DateTo.Value.Month;
+            paymentTwo.CalendarPeriodYear = DateTime.Now.Year;
+            paymentTwo.CalendarPeriodMonth = DateTime.Now.Month;
+            var paymentFundsOut = new List<PaymentFundsOut>{paymentOne,paymentTwo};
+            paymentFundsOutRepository.Setup(x => x.GetPaymentFundsOut(message.AccountId)).ReturnsAsync(paymentFundsOut);
+
+            expiredFunds.Setup(x => x.GetExpiringFunds(
+                It.Is<IDictionary<CalendarPeriod, decimal>>(c => c.First().Value.Equals(levyOne.FundsIn)), 
+                It.Is<IDictionary<CalendarPeriod, decimal>>(c => c.First().Value.Equals(paymentOne.FundsOut)), 
+                It.Is<IDictionary<CalendarPeriod, decimal>>(c=>c.Count.Equals(0)),
+                expiredFundsPeriod)).Returns(expiredFund);
+            
+            //Act
+            await handler.Handle(message, new TestableMessageHandlerContext(new MessageMapper()));
+
+            //Assert
+            expiredFundsRepository.Verify(x => x.GetDraft(message.AccountId), Times.Once);
```

Comment:
You should probably setup this method and pass the output to GetExpiringFunds then remove the verify...as it is you're not testing that the output is passed correctly.

Reply:
Updated